roles/redis/tasks/ensure_eda_workers_redis_connectivity.yml

```diff
@@ -0,0 +1,39 @@
+---
+- name: "Set initial facts"
+  ansible.builtin.set_fact:
+    _eda_workers_restart_required: false
+    _eda_workers_deployment_list:
+      - "eda-activation-worker"
+      - "eda-default-worker"
+      - "eda-scheduler"
+
+- name: "Get EDA worker deployments pods info"
+  kubernetes.core.k8s_info:
+    kind: Pod
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    label_selectors:
+      - "app.kubernetes.io/component={{ item }}"
+  loop: "{{ _eda_workers_deployment_list }}"
+  no_log: "{{ no_log }}"
+  register: _eda_worker_pod_info
+
+- name: "Check if restart is required"
+  vars:
+    _eda_worker_crashloopback_check: "{{ item.resources[0].status.containerStatuses[0].state.waiting.reason | default('Running') }}"
+    _eda_worker_completed_check: "{{ item.resources[0].status.containerStatuses[0].lastState.terminated.reason | default('Running') }}"
+  ansible.builtin.set_fact:
+    _eda_workers_restart_required: true
+  when: _eda_worker_crashloopback_check == 'CrashLoopBackOff' or _eda_worker_completed_check == 'Completed'
```

Comment:
Is this enough to detect the scenario? What if the worker has been restarted successfully?